app/actions/websocket/preferences.ts

```diff
@@ -2,11 +2,27 @@
 // See LICENSE.txt for license information.
 
 import {updateDmGmDisplayName} from '@actions/local/channel';
+import {appEntry} from '@actions/remote/entry';
 import {fetchPostById} from '@actions/remote/post';
+import {truncateCrtRelatedTables} from '@app/queries/servers/entry';
+import {popToRoot} from '@app/screens/navigation';
 import {Preferences} from '@constants';
 import DatabaseManager from '@database/manager';
 import {getPostById} from '@queries/servers/post';
-import {deletePreferences, differsFromLocalNameFormat} from '@queries/servers/preference';
+import {deletePreferences, differsFromLocalNameFormat, queryHasCRTChanged} from '@queries/servers/preference';
+
+async function handleCRTToggled(serverUrl: string) {
+    const currentServerUrl = await DatabaseManager.getActiveServerUrl();
+    const isSameServer = currentServerUrl === serverUrl;
+
+    const {error} = await truncateCrtRelatedTables(serverUrl);
+    if (!error) {
+        await appEntry(serverUrl);
+        if (isSameServer) {
+            popToRoot();
+        }
```

Comment:
do we need to await for `appEntry` before the pop to root?

The way I would address this is the following:
1. Emit some sort of event so that the Home screen reflects some UI that the account is switching between CRT / non-CRT
2. Pop to root and see the Home screen with the UI
3. Run app entry
4. If an error occurred show and handle the error
5. if no error emit the event to revert back the UI to the normal Home screen

cc: @matthewbirtch for input on my idea